curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -2324,6 +2324,7 @@ CURVEFS_ERROR DataCache::Flush(uint64_t inodeId, bool toS3) {
         S3ChunkInfo info;
         PrepareS3ChunkInfo(chunkId, offset, writeOffset, &info);
         inodeWrapper->AppendS3ChunkInfo(chunkIndex, info);
+        s3ClientAdaptor_->GetInodeCacheManager()->AddS3ChunkInfoMetric(1);
```

Comment:
I think it's better to pack addition and subtraction inside `InodeWrapper` because `InodeWrapper` holds all the s3chunkinfo, and your implementation adds interface to expose it.

For subtraction, a good place is InodeWrapper's destructor, because s3chunkinfo is expired there. Current implementation does this when an InodeWrapper is removed from  the CacheManager, it's not correct, InodeWrapper is shared which means it might still exist and costs memory.

For addition, I saw you do this in two places, one is getting an inode from metaserver, and another is after flushed a new object.

And the second one will cost memory twice, the extra one is cleared after updating to metaserver successful.

https://github.com/opencurve/curve/blob/cad667ccc3ad5a221ae3b43aad26e37c3122ff63/curvefs/src/client/inode_wrapper.h#L304

Reply:
fixed